cmd/metrics-server/app/options/options.go

```diff
@@ -63,6 +63,27 @@ type Options struct {
 	DeprecatedCompletelyInsecureKubelet bool
 }
 
+func (o *Options) Validate() []error {
+	errors := []error{}
+	if o.KubeletCAFile != "" && o.InsecureKubeletTLS {
+		errors = append(errors, fmt.Errorf("Cannot use both --kubelet-ca-file and --kubelet-insecure-tls"))
+	}
+
+	if o.KubeletClientKeyFile == "" && o.KubeletClientCertFile == "" {
+		errors = append(errors, fmt.Errorf("Need both --kubelet-client-key and --kubelet-client-certificate"))
+	}
+
+	if o.KubeletClientKeyFile == "" && o.DeprecatedCompletelyInsecureKubelet {
+		errors = append(errors, fmt.Errorf("Cannot use both --kubelet-client-key and --deprecated-kubelet-completely-insecure"))
+	}
+
+	if o.KubeletClientCertFile == "" && o.DeprecatedCompletelyInsecureKubelet {
+		errors = append(errors, fmt.Errorf("Cannot use both --kubelet-client-certificate and --deprecated-kubelet-completely-insecure"))
+	}
+
```

Comment:
There are some additional nontrivial cases. 

Please add case for mutually exclusive flags:
* `InsecureKubeletTLS` and `DeprecatedCompletelyInsecureKubelet`
* `KubeletCAFile` and `DeprecatedCompletelyInsecureKubelet`